rust/osm2lanes-web/src/main.rs

```diff
@@ -279,29 +279,28 @@ impl Component for App {
 }
 
 impl App {
-    fn calculate(
-        value: &str,
-        locale: &Locale,
-    ) -> Result<(Lanes, Tags), Result<(Lanes, String), String>> {
-        log::trace!("Calculate: {}", value);
-        match Tags::from_str(value) {
+    fn update_tags(&mut self) {
+        let value = &self.state.edit_tags;
+        let locale = &self.state.locale;
+        log::trace!("Update Tags: {}", value);
+        let calculate = match Tags::from_str(value) {
             Ok(tags) => match tags_to_lanes(&tags, locale, &TagsToLanesConfig::default()) {
                 Ok(lanes) => {
                     match lanes_to_tags(&lanes.lanes, locale, &LanesToTagsConfig::default()) {
                         Ok(tags) => Ok((lanes, tags)),
-                        Err(e) => Err(Ok((lanes, e.to_string()))),
+                        Err(e) => {
+                            if let RoadError::Warnings(warnings) = &e {
+                                Err(Ok((lanes, format!("{}\n{}", e, warnings))))
```

Comment:
Would this wind up repeating something? If you've got a before/after screenshot or example, that'd be helpful, or I can run locally and see...

Reply:
it does often wind-up repeating warnings, but this is important for debugging, because they do not have to be the same.

Warning generated in the first tags(input) to lanes,
are not the same as the warnings generated after tags(input) -> lanes -> tags -here-> lanes (rountrip testing the lanes to tags step).